cws2/migrations/0004_phono_systems.py

```diff
@@ -0,0 +1,266 @@
+                        ],
+                        help_text="Category for this Phone.",
+                        max_length=1,
+                        verbose_name="Category",
+                    ),
+                ),
+                (
+                    "ipa",
+                    models.CharField(
+                        help_text="IPA phonetic representation of this Phone.",
+                        max_length=12,
+                        verbose_name="IPA",
+                    ),
+                ),
+                (
+                    "xsampa",
+                    models.CharField(
+                        help_text="X-Sampa phonetic representation of this Phone.",
+                        max_length=12,
+                        verbose_name="X-Sampa",
+                    ),
+                ),
+                (
+                    "tags",
+                    django.contrib.postgres.fields.ArrayField(
+                        base_field=models.CharField(max_length=32),
+                        blank=True,
+                        default=list,
+                        help_text="Tags associated with this Phone.",
+                        size=None,
+                        verbose_name="Tags",
+                    ),
+                ),
+                (
+                    "archived_by",
+                    models.ForeignKey(
+                        blank=True,
+                        db_column="archived_by_user_id",
+                        db_index=False,
+                        null=True,
+                        on_delete=django.db.models.deletion.SET_NULL,
+                        related_name="archived_%(class)ss+",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+                (
+                    "created_by",
+                    models.ForeignKey(
+                        db_column="created_by_user_id",
+                        db_index=False,
+                        null=True,
+                        on_delete=django.db.models.deletion.SET_NULL,
+                        related_name="created_%(class)ss",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+                (
+                    "updated_by",
+                    models.ForeignKey(
+                        blank=True,
+                        db_column="updated_by_user_id",
+                        db_index=False,
+                        null=True,
+                        on_delete=django.db.models.deletion.SET_NULL,
+                        related_name="updated_%(class)ss+",
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+            ],
+            bases=(cws2.models.base.UUIDModel,
+                   cws2.models.base.OwnableModel, models.Model),
+        ),
+        migrations.AddConstraint(
+            model_name="phone",
+            constraint=models.UniqueConstraint(
```

Comment:
Constraints like this should be defined in model code too, in the model's Meta sub-class. If you do that too, Django can auto-generate the constraint in the migration from the code which is easier than having to manually write out the migration (:

```python
class MyModel(models.Model):
    class Meta:
        constraints = [
            models.UniqueConstraint(fields=["field1", "field2"], name="cws2_mymodel_unique_field1_field2"),
        ]
```